### PR TITLE
Add a parameter to allow disabling management of the ssh service

### PR DESCRIPTION
**Pull request by dfairhurst**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -580,6 +581,8 @@
   }
   validate_bool($purge_keys_real)
 
+  validate_bool($manage_service)
```
Comment (ghoneycutt) [inline manifests/init.pp]: Could you add the logic that allows for `'true'` as well as `true` ? See the section just above with purge_keys

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -3371,4 +3371,43 @@
...
+        let(:params) { { :manage_service => value } }
+        it do
+          expect {
+            should contain_class('ssh')
+          }.to raise_error(Puppet::Error,/is not a boolean/)
+        end
+      end
+    end
+
+    context "specified as true" do
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: test for boolean and string true. Here's an example

https://github.com/ghoneycutt/puppet-module-ntp/blob/master/spec/classes/init_spec.rb#L469

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -3371,4 +3371,43 @@
...
+      let(:params) { { :manage_service => true } }
+      it do
+        expect {
+          should contain_service('sshd_service')
+        }
+      end
+    end
+
+    context "specified as false" do
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: same for false

**Comment by dfairhurst** [comment]
Updated to accept string values as requested

**Comment by ghoneycutt** [comment]
Awesome work

**Comment by dfairhurst** [comment]
Thanks :) and thanks for the merge.

**Comment by ghoneycutt** [comment]
Released in v3.32.0

**Pull request merged**